catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -817,4 +782,50 @@ protected void validateGetResourceRequest(ResourceRequest getResourceRequest)
           "Cannot perform getResource with null attribute value or null attributeName, either passed in from endpoint, or as output from PreResourcePlugin");
     }
   }
+
+  private URI getDerivedResourceUri(Metacard metacard, String qualifier)
+      throws ResourceNotFoundException {
+    URI uriWithMatchingQualifier = null;
+    Attribute derivedResourcesAttribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+
+    if (derivedResourcesAttribute == null) {
+      throw new ResourceNotFoundException(
+          String.format("No derived resource uri's on metacard \"%s\"", metacard.getId()));
+    }
+
+    for (Serializable ser : derivedResourcesAttribute.getValues()) {
+      if (!(ser instanceof String)) {
+        continue;
+      }
+
+      URI uri;
+      try {
+        uri = new URI((String) ser);
+      } catch (URISyntaxException e) {
+        LOGGER.debug(
+            String.format(
+                "Found derived URI without the proper syntax \"%s\" on metacard \"%s\"",
+                ser, metacard.getId()));
+        continue;
+      }
+      if (!uri.getFragment().equals(qualifier)) {
+        continue;
+      }
+      if (uriWithMatchingQualifier != null) {
```

Comment:
❓ Will `uriWithMatchingQualifier` always be null at this point? It looks like it is assigned null at the beginning of this method, and is untouched until this check.

Reply:
It will be null the first time through but we're in a loop here so if we find a URI that contains the given qualifier we save it in uriWithMatchingQualifier. If we then come back through and find another one, then somehow the Uri got added to the metacard twice which is a no-no, and we'll then run through this if-statement. Looking back on it I should've put more effort into making this method readable.